service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/SpecimenMapperTest.java

```diff
@@ -0,0 +1,123 @@
+package uk.nhs.adaptors.gp2gp.ehr.mapper;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.hl7.fhir.dstu3.model.Observation;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.fhir.dstu3.model.ResourceType;
+import org.hl7.fhir.dstu3.model.Specimen;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.stubbing.Answer;
+
+import uk.nhs.adaptors.gp2gp.common.service.FhirParseService;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.diagnosticreport.ObservationMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.diagnosticreport.SpecimenMapper;
+import uk.nhs.adaptors.gp2gp.utils.ResourceTestFileUtils;
+
+@ExtendWith(MockitoExtension.class)
+public class SpecimenMapperTest {
+
+    private static final String TEST_FILE_DIRECTORY = "/ehr/mapper/specimen/";
+    private static final String DIAGNOSTIC_REPORT_DATE = "2020-10-12";
+    private static final String INPUT_OBSERVATION_RELATED_TO_SPECIMEN = "input-observation-1.json";
+    private static final String INPUT_OBSERVATION_NOT_RELATED_TO_SPECIMEN = "input-observation-2.json";
+
+    private SpecimenMapper specimenMapper;
+
+    @Mock
+    private MessageContext messageContext;
+    @Mock
+    private IdMapper idMapper;
+    @Mock
+    private ObservationMapper observationMapper;
+    private List<Observation> observations;
+
+    @BeforeEach
+    public void setUp() throws IOException {
+        lenient().when(messageContext.getIdMapper()).thenReturn(idMapper);
+        lenient().when(idMapper.getOrNew(any(ResourceType.class), anyString())).thenAnswer(mockId());
+        lenient().when(idMapper.get(any(Reference.class))).thenAnswer(mockReference());
+
+        observations = Arrays.asList(
+            parseObservation(INPUT_OBSERVATION_RELATED_TO_SPECIMEN),
+            parseObservation(INPUT_OBSERVATION_NOT_RELATED_TO_SPECIMEN)
+        );
```

Comment:
The community seems to be moving towards reserving `Arrays.asList()` for when you're actually working with arrays, and to prefer `List.of()`.
```suggestion
        observations = List.of(
            parseObservation(INPUT_OBSERVATION_RELATED_TO_SPECIMEN),
            parseObservation(INPUT_OBSERVATION_NOT_RELATED_TO_SPECIMEN)
        );
```

An alternative, since you're doing the same thing to each (both) constant:
```suggestion
        observations = Stream.of(INPUT_OBSERVATION_RELATED_TO_SPECIMEN, INPUT_OBSERVATION_NOT_RELATED_TO_SPECIMEN)
            .map(this::parseObservation)
            .collect(Collectors.toList());
```

Reply:
I chose first option, because parseObservation is throwing exception.